apps/talk_signaling.sh

```diff
@@ -0,0 +1,403 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+# shellcheck disable=2034,2059
+true
+# shellcheck source=lib.sh
+NC_UPDATE=1 && TURN_INSTALL=1 . <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+unset NC_UPDATE
+unset TURN_INSTALL
+
+# Check for errors + debug code and abort if something isn't right
+# 1 = ON
+# 0 = OFF
+DEBUG=0
+debug_mode
+
+# Must be root
+root_check
+
+# Must be 20.04
+if ! version 20.04 "$DISTRO" 20.04.6
+then
+msg_box "Your current Ubuntu version is $DISTRO but must be between 20.04 - 20.04.6 to install Talk"
+msg_box "Please contact us to get support for upgrading your server:
+https://www.hanssonit.se/#contact
+https://shop.hanssonit.se/"
+exit
+fi
+
+# Nextcloud 13 is required.
+lowest_compatible_nc 19
+
+####################### TALK (COTURN)
+
+# Check if Nextcloud is installed with TLS
+check_nextcloud_https "Nextclod Talk"
+
+# Check if talk/spreed is already installed
+print_text_in_color "$ICyan" "Checking if Talk is already installed..."
+if [ -n "$(occ_command_no_check config:app:get spreed turn_servers | sed 's/\[\]//')" ] || is_this_installed coturn
+then
+    choice=$(whiptail --radiolist "It seems like 'Nextcloud Talk' is already installed.\nChoose what you want to do.\nSelect by pressing the spacebar and ENTER" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    "Uninstall Nextcloud Talk" "" OFF \
+    "Reinstall Nextcloud Talk" "" ON 3>&1 1>&2 2>&3)
+    
+    case "$choice" in
+        "Uninstall Nextcloud Talk")
+            print_text_in_color "$ICyan" "Uninstalling Nextcloud Talk and resetting all settings..."
+            occ_command_no_check config:app:delete spreed stun_servers
+            occ_command_no_check config:app:delete spreed turn_servers
+            occ_command_no_check config:app:delete spreed signaling_servers
+            occ_command_no_check app:remove spreed
+            rm -rf \
+              "$TURN_CONF" \
+              "$SIGNALING_SERVER_CONF" \
+              /etc/nats \
+              /etc/janus \
+              /etc/apt/trusted.gpg.d/morph027-janus.asc \
+              /etc/apt/trusted.gpg.d/morph027-nats-server.asc \
+              /etc/apt/trusted.gpg.d/morph027-nextcloud-spreed-signaling.asc \
+              /etc/apt/sources.list.d/morph027-nextcloud-spreed-signaling.list\
+              /etc/apt/sources.list.d/morph027-janus.list \
+              /etc/apt/sources.list.d/morph027-nats-server.list
+            apt-get purge coturn nats-server janus nextcloud-spreed-signaling -y
+            apt autoremove -y
+            check_command systemctl disable signaling
```

Comment:
this will fail if it is already uninstalled

Reply:
Please remove

Reply:
ive changed it